package/MDAnalysis/core/universe.py

```diff
@@ -124,54 +126,59 @@ def __init__(self, *args, **kwargs):
             self.atoms = None
             return
 
-        self.filename = args[0]
         coordinatefile = args[1:]
 
-        topology_format = kwargs.pop('topology_format', None)
-
-        if len(args) == 1:
-            # special hacks to treat a coordinate file as a coordinate AND topology file
-            if kwargs.get('format', None) is None:
-                kwargs['format'] = topology_format
-            elif topology_format is None:
-                topology_format = kwargs.get('format', None)
-
-            # if passed a Reader, use that
-            fmt = kwargs.get('format', None)
+        # if we're given a Topology object, we don't need to parse anything
+        if isinstance(args[0], Topology):
+            self._topology = args[0]
+            self.filename = None
```

Comment:
Maybe topology should remember the filename in the Topology object, so here self.filename = topology.filename.  It will be annoying if somewhere down the line we've lost memory of what file this came from